src/equations/shallow_water_two_layer_1d.jl

```diff
@@ -0,0 +1,586 @@
+  h1_rr, h2_rr = waterheight(u_rr, equations)
+  c_ll = sqrt(equations.gravity * (h1_ll + h2_ll))
+  c_rr = sqrt(equations.gravity * (h1_rr + h2_rr))
+
+  return (max(abs(Um_ll) + c_ll, abs(Um_rr) + c_rr))
+end
+
+
+# Specialized `DissipationLocalLaxFriedrichs` to avoid spurious dissipation in the bottom 
+# topography
+@inline function (dissipation::DissipationLocalLaxFriedrichs)(u_ll, u_rr,
+    orientation_or_normal_direction, equations::ShallowWaterTwoLayerEquations1D)
+  λ = dissipation.max_abs_speed(u_ll, u_rr, orientation_or_normal_direction, equations)
+  diss = -0.5 * λ * (u_rr - u_ll)
+  return SVector(diss[1], diss[2], diss[3], diss[4], zero(eltype(u_ll)))
+end
+
+
+# Absolute speed of the barotropic mode
+@inline function max_abs_speeds(u, equations::ShallowWaterTwoLayerEquations1D)
+  h1, h2 = waterheight(u, equations)
+  
+  # Calculate averaged velocity of both layers
+  v = (u[2] + u[4]) / (h1 + h2)
+  c = sqrt(equations.gravity * (h1 + h2))
+
+  return (abs(v) + c)
+end
+
+
+# Helper function to extract the velocity vector from the conservative variables
+@inline function velocity(u, equations::ShallowWaterTwoLayerEquations1D)
+  h1, h1_v1, h2, h2_v2, _ = u
+
+  v1 = h1_v1 / h1
+  v2 = h2_v2 / h2
+  return v1, v2
+end
+
+
+# Convert conservative variables to primitive
+@inline function cons2prim(u, equations::ShallowWaterTwoLayerEquations1D)
+  h1, _, h2, _, b = u
+
+  H2 = h2 + b
+  H1 = h2 + h1 + b
+  v1, v2 = velocity(u, equations)
+  return SVector(H1, v1, H2, v2, b)
+end
+
+
+# Convert conservative variables to entropy variables
+# Note, only the first four are the entropy variables, the fifth entry still just carries the 
+# bottom topography values for convenience
+@inline function cons2entropy(u, equations::ShallowWaterTwoLayerEquations1D)
+  h1, h1_v1, h2, h2_v2, b = u
+  ρ1 = equations.rho1
+  ρ2 = equations.rho2
+
+  v1, v2 = velocity(u, equations)
+
+  w1 = ρ1 * (equations.gravity * (h1 + h2 + b) - 0.5 * v1^2)
+  w2 = ρ1 * v1
+  w3 = ρ2 * (equations.gravity * (equations.r * h1 + h2 + b) - 0.5 * v2^2)
+  w4 = ρ2 * v2
+  return SVector(w1, w2, w3, w4, b)
+end
+
+
+# Convert primitive to conservative variables
+@inline function prim2cons(prim, equations::ShallowWaterTwoLayerEquations1D)
+  H1, v1, H2, v2, b = prim
+
+  h2    = H2 - b
+  h1    = H1 - h2 - b
+  h1_v1 = h1 * v1
+  h2_v2 = h2 * v2
+  return SVector(h1, h1_v1, h2, h2_v2, b)
+end
+
+
+@inline function waterheight(u, equations::ShallowWaterTwoLayerEquations1D)
+  return u[1], u[3]
+end
+
+
+# Entropy function for the shallow water equations is the total energy
+@inline entropy(cons, equations::ShallowWaterTwoLayerEquations1D) = energy_total(cons, equations)
+
+
+# Calculate total energy for a conservative state `cons`
+@inline function energy_total(cons, equations::ShallowWaterTwoLayerEquations1D)
+  h1, h2, h1_v1, h2_v2, b = cons
+  # Set new variables for better readability
+  g = equations.gravity
+  ρ1 = equations.rho1
+  ρ2 = equations.rho2
```

Comment:
@sloede hates non-ASCII code

Reply:
I see your point. I will change it to 
```suggestion
  rho1 = equations.rho1
  rho2 = equations.rho2
```
instead.